inotify/src/lib.rs

```diff
@@ -264,10 +265,17 @@ impl Inotify {
     /// [`Inotify::add_watch`]: struct.Inotify.html#method.add_watch
     /// [`Event`]: struct.Event.html
     pub fn rm_watch(&mut self, wd: WatchDescriptor) -> io::Result<()> {
-        let result = unsafe { ffi::inotify_rm_watch(self.0, wd.0) };
+        let result: c_int;
+        if self.0 == wd.1 {
+            result = unsafe { ffi::inotify_rm_watch(self.0, wd.0) }; 
+        }
+        else {
+            result = -2;
```

Comment:
I don't like how you use `result` here, for the following reasons:
1. Using an integer variable to express the error condition is not type-safe. The only reason we have that here at all, is because of the underlying inotify API.
2. You're overloading the meaning of `result`. Before it was about one thing only (the return value from `inotify_rm_watch`), now there's a second concern (whether the `WatchDescriptor` is valid) mixed into it.

To be clear, I believe this works (haven't tested it), but the code is not as clear as it could be. I'd just leave the two things separate: Check the watch descriptor at the beginning and do an early return if it's invalid. Leave the rest of the method as it is.